extensions/ql-vscode/src/distribution.ts

```diff
@@ -354,26 +358,46 @@ class ExtensionSpecificDistributionManager {
     }
   }
 
+  /**
+   * Get the name of the codeql cli installation we prefer to install, based on our current platform.
+   */
+  private getRequiredAssetName(): string {
+    if (os.platform() === 'linux') return 'codeql-linux64.zip';
+    if (os.platform() === 'darwin') return 'codeql-osx64.zip';
+    if (os.platform() === 'win32') return 'codeql-win64.zip';
+    return 'codeql.zip';
+  }
+
   private async getLatestRelease(): Promise<Release> {
-    return await this.createReleasesApiConsumer().getLatestRelease(
+    const requiredAssetName = this.getRequiredAssetName();
+    logger.log(`Searching for latest release including ${requiredAssetName}. `);
+    const release = await this.createReleasesApiConsumer().getLatestRelease(
       this._versionRange,
       this._config.includePrerelease,
       release => {
-        // FIXME: Look for platform-specific codeql distribution if available
-        // https://github.com/github/vscode-codeql/issues/417
-        const matchingAssets = release.assets.filter(asset => asset.name === 'codeql.zip');
+        const matchingAssets = release.assets.filter(asset => asset.name === requiredAssetName);
         if (matchingAssets.length === 0) {
           // For example, this could be a release with only platform-specific assets.
-          logger.log("INFO: Ignoring a release with no assets named codeql.zip");
+          logger.log(`INFO: Ignoring a release with no assets named ${requiredAssetName}`);
           return false;
         }
         if (matchingAssets.length > 1) {
-          logger.log("WARNING: Ignoring a release with more than one asset named codeql.zip");
+          logger.log(`WARNING: Ignoring a release with more than one asset named ${requiredAssetName}`);
           return false;
         }
         return true;
       }
     );
+    // Actually filter assets to the unique one that we require.
+    release.assets = release.assets.filter(asset => asset.name === requiredAssetName);
```

Comment:
Do we want to first look for the platform specific asset, and only if that one isn't found, we then look for the generic asset?

Reply:
I believed (correct me if I'm wrong @hmakholm) the plan was to, for the time being, make releases in pairs, one (call it `v3.4.5`) which contains the all-platform `codeql.zip`, and another (named something like `v3.4.5-platform`) which contains separate platform-specific releases. I believe @henrymercer's latest revision to `ReleasesApiConsumer.getLatestRelease` will find the latest release _which satisfies `additionalCompatibilityCheck`_. So when `v3.4.5` and `v3.4.5-platform` have been released, this code, on for example linux, will download `v3.4.5-platform` because it will be the latest release _that contains `codeql-linux64.zip`_.

However, now that I think about it, I'm still not sure I'm convinced this plan is watertight. With the version of `distribution.ts` that we've had for a while, namely https://github.com/github/vscode-codeql/blob/7d23a833b178690eede71aa5e292c3c4b369ccab/extensions/ql-vscode/src/distribution.ts is it not the case that the extension will take whichever of `v3.4.5-platform` and `v3.4.5` has been released chronologically most recently? And if that is true, won't there be some moment, no matter how we order their release, when `v3.4.5-platform` is the most recent, and the extension has a chance of downloading a release with multiple release assets?

Reply:
@jcreedcmu, I must addmt I have tuned a bit out of the discussion, so personally my plan for the time being is to wait until the next release draws near and then ask #codeql-extensions what a timing and naming strategy would be that (a) works with all supported version of the extension, and (b) under that constraint, gives the best experience for users of the _latest_ version of the extension.

Reply:
(Or at least it is if we are committed to a cli release schedule that always delivers platform specific zips in some form - merging this PR amounts to trusting that commitment has been made)

Reply:
@hmakholm that is entirely reasonable; my earlier worry was that a naming strategy that satisfied both (a) and "involves any platform-specific assets at all" _didn't exist_, but I think now you've convinced me that it does.